clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -468,6 +468,55 @@ def BrOp : CIR_Op<"br",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// UnaryOp
+//===----------------------------------------------------------------------===//
+
+def UnaryOpKind_Inc   : I32EnumAttrCase<"Inc",   1, "inc">;
```

Comment:
do we not need to differentiate pre and post increment?  They have sufficiently different semantics.

Reply:
The pre/post semantics are captured in the CIR itself by where the `cir.unary` op appears in the control flow.